site/app/controllers/PollController.php

```diff
@@ -398,4 +399,64 @@ public function viewResults($poll_id) {
             )
         );
     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/polls/getPollExport", methods={"GET"})
+     * @AccessControl(role="INSTRUCTOR")
+     * @return JsonResponse
```

Comment:
This isn't asking expressly to change anything, just for clarification for myself to understand this code / purpose.

Reply:
It should just be for the ajax download. If there's a better way to make this function to reflect that, let me know and I can change it.

Reply:
You could mimic some of the functions in [MiscController](https://github.com/Submitty/Submitty/blob/master/site/app/controllers/MiscController.php) (like displayFile) where you turn off header/footer, manually set some headers to register it as a download, and then use `$this->core->getOutput()->renderString` to output the JSON. The JS could then be removed and replaced with a regular `<a>` tag. If you want to do it, that's fine, can also hold off and I think first put in place a generic `DownloadResponse` that makes doing this easier around the system.

Let me know which you'd prefer to do here.

Reply:
I'll work on making a DownloadResponse. Seems like it will be useful.

Reply:
For the DownloadResponse, would it make sense for it to just take in a path to a file that will be downloaded? Then it can grab the mime info from the file and download it that way.

Reply:
Let's not do the DownloadResponse within this PR, and instead do it separately (after this is merged in the next day or so) as then that separate PR can address all of the places such a new class would benefit, and the class can be built to cater to all of them from the get-go. For doing that, can you spin this conversation into a new issue and then also include all of the functions that would benefit from this new DownloadResponse? This should then give us a good idea of where we would expect to utilize this to users. I expect that some of the methods would utilize a file (and so would use the php function `readfile` as well) while others are just going to output some string buffer (e.g. this method).

Reply:
Roger that. I've updated this PR to use the same method as MiscController::displayFile which should be easy to change over once the new Response type is made.